frontend/test/lit/test_detensorize.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# RUN: %PYTHON %s | FileCheck %s
+
+# pylint: disable=missing-module-docstring,missing-function-docstring
+
+from catalyst import qjit
+from catalyst.debug import get_compilation_stage
+
+
+# CHECK-LABEL: public @jit_a_plus_b_times_2
+@qjit(keep_intermediate=True)
+def a_plus_b_times_2(a, b):
+    # CHECK: %extracted
+    # CHECK: tensor.extract
+    # CHECK: arith.addf
+    # CHECK-NOT: linalg.generic
+    c = a + b
+    # CHECK: tensor.from_elements
+    return c + c
+
+
+a_plus_b_times_2(1.0, 2.0)
+a_plus_b_times_2.workspace.cleanup()
```

Comment:
Maybe a little bit weird, I would expect the cleanup to be after `get_compilation_stage`. There's technically a string stored in the object `a_plus_b_times_2`, but this is something that we might change in the future.

Reply:
The reason I wanted to cleanup was to avoid the following situation: 
- first run the test and because of  `keep_intermediate=True` a directory `a_plus_b_times_2` is created. 
- run the test again and a directory `a_plus_b_times_2_1` is created.
- I to avoid reading from `a_plus_b_times_2`, when the latest test IR is in `a_plus_b_times_2_1`, I deleted `a_plus_b_times_2`.
Your suggestion should prevent that too, cleaning preemptively after oneself.